.github/workflows/deploy-integrations-staging.yml

```
@@ -43,7 +43,7 @@ jobs:
           linear_secrets: '' # secrets should already exst
           slack_secrets: '' # secrets should already exist
           charts_secrets: '--secrets QUICKCHARTS_API_KEY=${{ secrets.QUICKCHARTS_API_KEY }}'
-          whatsapp_secrets: '--secrets CLIENT_ID=${{ secrets.STAGING_WHATSAPP_CLIENT_ID }} --secrets CLIENT_SECRET=${{ secrets.STAGING_WHATSAPP_CLIENT_SECRET }} --secrets ACCESS_TOKEN=${{ secrets.STAGING_WHATSAPP_ACCESS_TOKEN }} --secrets NUMBER_PIN=${{ secrets.STAGING_WHATSAPP_NUMBER_PIN }}'
+          whatsapp_secrets: '--secrets CLIENT_ID=${{ secrets.STAGING_WHATSAPP_CLIENT_ID }} --secrets CLIENT_SECRET=${{ secrets.STAGING_WHATSAPP_CLIENT_SECRET }} --secrets ACCESS_TOKEN=${{ secrets.STAGING_WHATSAPP_ACCESS_TOKEN }} --secrets NUMBER_PIN=${{ secrets.STAGING_WHATSAPP_NUMBER_PIN }} --secrets SEGMENT_KEY=${{ secrets.PRODUCTION_WHATSAPP_SEGMENT_KEY }}'
```

As this is the staging workflow, the secret should be prefixed with `STAGING_` (unless we want to use the same for both environments, in which case we could rename it to just `WHATSAPP_SEGMENT_KEY`).

```suggestion
          whatsapp_secrets: '--secrets CLIENT_ID=${{ secrets.STAGING_WHATSAPP_CLIENT_ID }} --secrets CLIENT_SECRET=${{ secrets.STAGING_WHATSAPP_CLIENT_SECRET }} --secrets ACCESS_TOKEN=${{ secrets.STAGING_WHATSAPP_ACCESS_TOKEN }} --secrets NUMBER_PIN=${{ secrets.STAGING_WHATSAPP_NUMBER_PIN }} --secrets SEGMENT_KEY=${{ secrets.STAGING_WHATSAPP_SEGMENT_KEY }}'
```

---

the action.yml already had whatsapp secrets https://github.com/botpress/botpress/blob/7bd5ead89f2aabc409c6573146493cf651452c9b/.github/actions/deploy-integrations/action.yml#L129-L130 

Shouldn't this suffice?

---

and also, nice catch on the staging variable name!

---

You're right, I was confused, it's already configured 👍